debug! #327

webknjaz opened the pull request:
PR #327 has a failing CI. This is an attempt to verify whether it's failing on the default branch or the failure is related to the patch (the pins update).

webknjaz committed:
debug!

webknjaz committed:
debug! pins

webknjaz committed:
Revert "debug! pins"

This reverts commit 6ef6be9bfcc43a3186fbd6720ef0a38e412b6fb5.

webknjaz committed:
debug! py3.9

kytta commented:
Ah shucks, just noticed there's a different PR for this issue. See https://github.com/pypa/gh-action-pypi-publish/pull/327#issuecomment-2609037748; I propose we pin the Python to a newer version than 3.9 (which is EOL this year), and we upgrade the setuptools version that is used as a build system. Maybe don't even hard-pin it, but use a `>=`?

webknjaz committed:
Revert "debug! py3.9"

This reverts commit 688a8c45ff0c138ef01a74653903c28fd4a56e18.

webknjaz committed:
debug! bump setuptools to v75.8.0

webknjaz commented:
No, I want reproducibility. If anything, it's underpinned. Pinning the OS will also ensure a static major version of Python, which is enough.

webknjaz committed:
🧪 Run smoke tests against Ubuntu 24 and 22

They are pinned instead of using `-latest` in the interest of better
reproducibility in the CI.
